test/e2e/e2e_test.go

```diff
@@ -20,86 +20,111 @@ package e2e
+	defaultNamespace                            = "default"
 )
 
 var (
-	allNamespacesReadAccessPath, singleNamespaceReadAccessPath string
+	allNamespacesReadAccessTokenFile,
+	singleNamespaceReadAccessTokenFile,
+	allNamespacesAdminAccessTokenFile,
+	allNamespacesImpersonateAccessTokenFile,
+	certFile string
 )
 
 func init() {
-	tokensFolder := os.Getenv("SA_TOKEN_PATH")
-	if len(tokensFolder) == 0 {
-		tokensFolder = defTokenFolder
+	certPath := os.Getenv("SSL_CERT_PATH")
+	if len(certPath) == 0 {
+		certPath = defaultCertPath
 	}
-	allNamespacesReadAccessPath = path.Join(tokensFolder, allNamespacesReadAccessTknFile)
-	singleNamespaceReadAccessPath = path.Join(tokensFolder, singleNamespaceReadAccessTknFile)
+	certFile = path.Join(certPath, certFileName)
+
+	tokenPath := os.Getenv("SA_TOKEN_PATH")
+	if len(tokenPath) == 0 {
+		tokenPath = defaultTokenPath
+	}
+	allNamespacesReadAccessTokenFile = path.Join(tokenPath, allNamespacesReadAccessTokenFileName)
+	singleNamespaceReadAccessTokenFile = path.Join(tokenPath, singleNamespaceReadAccessTokenFileName)
+	allNamespacesAdminAccessTokenFile = path.Join(tokenPath, allNamespacesAdminAccessTokenFileName)
+	allNamespacesImpersonateAccessTokenFile = path.Join(tokenPath, allNamespacesImpersonateAccessTokenFileName)
 }
 
 func TestTaskRun(t *testing.T) {
 	ctx := context.Background()
-	tr := new(v1beta1.TaskRun)
-	b, err := ioutil.ReadFile("testdata/taskrun.yaml")
+	tr := new(tektonv1beta1.TaskRun)
+	b, err := os.ReadFile("testdata/taskrun.yaml")
 	if err != nil {
-		t.Fatalf("ioutil.Readfile: %v", err)
+		t.Fatalf("Error reading file: %v", err)
 	}
 	if err := yaml.UnmarshalStrict(b, tr); err != nil {
-		t.Fatalf("yaml.Unmarshal: %v", err)
+		t.Fatalf("Erro unmarshalling: %v", err)
 	}
 
-	c := clientTekton(t)
+	tc := tektonClient(t)
 
 	// Best effort delete existing Run in case one already exists.
-	_ = c.TaskRuns(ns).Delete(ctx, tr.GetName(), metav1.DeleteOptions{})
+	_ = tc.TaskRuns(defaultNamespace).Delete(ctx, tr.GetName(), metav1.DeleteOptions{})
 
-	tr, err = c.TaskRuns(ns).Create(ctx, tr, metav1.CreateOptions{})
+	tr, err = tc.TaskRuns(defaultNamespace).Create(ctx, tr, metav1.CreateOptions{})
 	if err != nil {
-		t.Fatalf("Create: %v", err)
+		t.Fatalf("Error creating TaskRun: %v", err)
 	}
-	t.Logf("Created TaskRun %s", tr.GetName())
+
+	gc, _ := resultsClient(t, allNamespacesReadAccessTokenFile, nil)
+
+	var resName, recName string
 
 	// Wait for Result ID to show up.
-	t.Run("Result ID", func(t *testing.T) {
+	t.Run("check annotations", func(t *testing.T) {
 		if err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (done bool, err error) {
-			tr, err := c.TaskRuns(ns).Get(ctx, tr.GetName(), metav1.GetOptions{})
-			t.Logf("Get: %+v %v", tr.GetName(), err)
+			tr, err := tc.TaskRuns(defaultNamespace).Get(ctx, tr.GetName(), metav1.GetOptions{})
 			if err != nil {
-				return false, nil
+				t.Fatalf("Error getting TaskRun: %v", err)
```

Comment:
Shouldn't we add `return false, err` here. Also line https://github.com/tektoncd/results/pull/418/files#diff-d5d12b6ea07b0a78c9ad1987e1486c565ecf16839f77a93e91b09475cf6c0e99R131 to be executed we should throw `err` I think.

Reply:
This is now running inside a test function that has no return. `t.Fatalf` is the right thing to do.

Reply:
This is the return statement of the callback function to `PollImmediate`. I guess calling `t.Fatal()` here would mark the test as failed before even waiting for the timeout.